src/common/lru_cache.h

```diff
@@ -356,6 +356,195 @@ std::shared_ptr<CacheMetrics>
+    std::unordered_map<K, typename std::list<K>::iterator> cache_;
+
+    // cache related metric data
+    std::shared_ptr<CacheMetrics> cacheMetrics_;
+};
+
+template <typename K, typename KeyTraits>
+std::shared_ptr<CacheMetrics>
+    SglLRUCache<K, KeyTraits>::GetCacheMetrics() const {
+    return  cacheMetrics_;
+}
+
+template <typename K, typename KeyTraits>
+uint64_t SglLRUCache<K, KeyTraits>::Size() {
+    ::curve::common::WriteLockGuard guard(lock_);
+    return ll_.size();
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::Put(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    PutLocked(key);
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::GetBack(K *value) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    if (ll_.empty()) {
+        LOG(INFO) << "cache is empty.";
+        return;
+    }
+    *value = ll_.back();
+    return;
+}
+
+template <typename K, typename KeyTraits>
+bool SglLRUCache<K, KeyTraits>::IsCached(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    auto iter = cache_.find(key);
+    if (iter == cache_.end()) {
+        if (cacheMetrics_ != nullptr) {
+            cacheMetrics_->OnCacheMiss();
+        }
+        return false;
+    }
+
+    if (cacheMetrics_ != nullptr) {
+        cacheMetrics_->OnCacheHit();
+    }
+
+    // update the position of the target item in the list
+    MoveToFront(iter->second);
+    return true;
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::Remove(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    RemoveLocked(key);
+}
+
+template <typename K, typename KeyTraits>
+bool SglLRUCache<K, KeyTraits>::PutLocked(const K &key) {
+    auto iter = cache_.find(key);
+
+    // delete the old value if already exist
+    if (iter != cache_.end()) {
+        RemoveElement(iter->second);
+    }
+    // put new value
+    ll_.push_front(key);
+    cache_[key] = ll_.begin();
+    if (cacheMetrics_ != nullptr) {
+        cacheMetrics_->UpdateAddToCacheCount();
+        cacheMetrics_->UpdateAddToCacheBytes(KeyTraits::CountBytes(key));
+    }
+    if (maxCount_ != 0 && ll_.size() > maxCount_) {
+        return RemoveOldest();
+    }
+    return false;
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::RemoveLocked(const K &key) {
+    auto iter = cache_.find(key);
+    if (iter != cache_.end()) {
+        RemoveElement(iter->second);
+    }
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::MoveToFront(
+    const typename std::list<K>::iterator &elem) {
+    K tmp = *elem;
+    ll_.erase(elem);
+    ll_.push_front(tmp);
+    cache_[tmp] = ll_.begin();
+}
+
+template <typename K, typename KeyTraits>
+bool SglLRUCache<K, KeyTraits>::RemoveOldest() {
+    if (ll_.begin() != ll_.end()) {
+        RemoveElement(--ll_.end());
```

Comment:
add a "return true" in line 333  for me,  there is a bug there

Reply:
fix